backend/logic/courses/courses.js

```diff
@@ -1,19 +1,72 @@
-const mysql = require('../../sql/connection');
+const mysql = require("../../sql/connection");
+const format = require("../../validation/format")
 
-async function queryCourseByTag(tag) {
-    if (!tag) {
-        throw Error("Undefined tag")
-    }
+var queryCourseByTag = async function queryCourseByTag(tag) {
+  if (!tag) {
+    throw Error("Undefined tag")
+  }
+
+  let connection = await mysql.getNewConnection();
+  try {
+    let courses = await connection.query("SELECT course_code FROM course_tag WHERE course_tag.tag_name LIKE ?", tag);
+    connection.release();
+    return (JSON.parse(JSON.stringify(courses)));
+  } catch (error) {
+    connection.release();
+    console.error(error);
```

Comment:
maybe it's not necessary to log the error now, since when you are throwing it, the message is already displayed in the output, and it might already in the log (not sure about this).

Reply:
I don't see the log if I don't do a console.error. Only the status of the endpoint call is shown

Reply:
ok nvm then